cmd/arcaflow/main.go

```diff
@@ -125,31 +114,44 @@ Options:
 	var err error
 
 	requiredFiles := map[string]string{
-		RequiredFileKeyConfig:   configFile,
-		RequiredFileKeyInput:    input,
 		RequiredFileKeyWorkflow: workflowFile,
 	}
 
-	fileCtx, err := loadfile.NewFileCacheUsingContext(dir, requiredFiles)
-	if err != nil {
-		flag.Usage()
-		tempLogger.Errorf("context path resolution failed %s (%v)", dir, err)
-		os.Exit(ExitCodeInvalidData)
+	if len(configFile) != 0 {
+		requiredFiles[RequiredFileKeyConfig] = configFile
 	}
 
-	configFilePath, err := fileCtx.AbsPathByKey(RequiredFileKeyConfig)
+	if len(input) != 0 {
+		requiredFiles[RequiredFileKeyInput] = input
+	}
+
+	fileCtx, err := loadfile.NewFileCacheUsingContext(dir, requiredFiles)
 	if err != nil {
-		tempLogger.Errorf("Unable to find configuration file %s (%v)", configFile, err)
+		tempLogger.Errorf("Context path resolution failed %s (%v)", dir, err)
 		flag.Usage()
 		os.Exit(ExitCodeInvalidData)
 	}
 
 	var configData any
-	configData, err = loadYamlFile(configFilePath)
-	if err != nil {
-		tempLogger.Errorf("Failed to load configuration file %s (%v)", configFile, err)
-		flag.Usage()
-		os.Exit(ExitCodeInvalidData)
+	if len(configFile) == 0 {
+		if err := yaml.Unmarshal([]byte(defaultConfig), &configData); err != nil {
+			tempLogger.Errorf("Failed to load default configuration", err)
+			os.Exit(ExitCodeInvalidData)
+		}
```

Comment:
This is slightly weird -- every time we run, we're dynamically creating the `configData` object by parsing a static string.  Rather than declaring a string containing YAML, we _could_ just declare a native object (map?) called `configData` and initialize it appropriately...which we could then overwrite if the length of `configFile` is zero...which would simply this code a bit.

Reply:
I follow your point, but this gets outside of my comfort zone with Go. Statically declaring a nested map is... confusing. So I went the "easier" route of unmarshalling the string.